api/environments/job_handler.go

```diff
@@ -137,6 +166,38 @@ func (eh EnvironmentHandler) getBatches(appName, envName, jobComponentName strin
 	return eh.getScheduledBatchSummaryList(radixBatches), nil
 }
 
+// StopBatch Stop batch by name
+func (eh EnvironmentHandler) StopBatch(appName, envName, jobComponentName, batchName string) error {
+	batch, err := eh.getRadixBatch(appName, envName, jobComponentName, batchName, kube.RadixBatchTypeBatch)
+	if err != nil {
+		return err
+	}
+
+	if batch.Status.Condition.Type != radixv1.BatchConditionTypeActive && batch.Status.Condition.Type != radixv1.BatchConditionTypeWaiting {
+		return nil // invalid batch status state
+	}
+
+	// create filter for jobs not to be stopped
+	jobStatus := slice.FindAll(batch.Status.JobStatuses, func(js radixv1.RadixBatchJobStatus) bool {
+		return js.Phase != radixv1.BatchJobPhaseActive && js.Phase != radixv1.BatchJobPhaseWaiting && js.Phase != ""
+	})
+
+	didChange := false
+	for idx, job := range batch.Spec.Jobs {
+		if slice.FindIndex(jobStatus, func(js radixv1.RadixBatchJobStatus) bool { return js.Name == job.Name }) == -1 {
+			batch.Spec.Jobs[idx].Stop = radixutils.BoolPtr(true)
+			didChange = true
+		}
+	}
```

Comment:
Create common function/logic to check if a job is stoppable, and use this in both StopBatch and StopJob

Reply:
fixed